api/src/main/java/org/pmiops/workbench/workspaces/WorkspacesController.java

```diff
@@ -545,14 +510,20 @@ private void setDbWorkspaceFields(
 
     dbWorkspace.setBillingMigrationStatusEnum(BillingMigrationStatus.NEW);
 
-    updateWorkspaceBillingAccount(dbWorkspace, body.getWorkspace().getBillingAccountName());
+    try {
+      workspaceService.updateWorkspaceBillingAccount(
```

Comment:
Eventually we'll also want some validation here that the creator still has free tier credit available. Do you know whether this is tracked?

Reply:
I thought we were going to allow users to continue creating workspaces in the free tier so I left it as is. I can ask Karthik if we want some UI treatment around how we handle the expired free tier case.